packages/angular/src/migrations/update-16-0-0/remove-karma-defaults.ts

```diff
@@ -0,0 +1,62 @@
+import type { Tree } from '@nx/devkit';
+import {
+  getProjects,
+  readNxJson,
+  updateNxJson,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+
+const POSSIBLE_GENERATORS = [
+  '@nrwl/angular:application',
+  '@nrwl/angular:library',
+  '@nrwl/angular:host',
+  '@nrwl/angular:remote',
+];
+
+export default async function removeKarmaDefaults(tree: Tree) {
+  const nxJson = readNxJson(tree);
+
+  if (nxJson.generators) {
+    const generators = Object.entries(nxJson.generators);
+    for (const [generatorName, generatorDefaults] of generators) {
+      if (!POSSIBLE_GENERATORS.includes(generatorName)) {
+        continue;
+      }
+
+      if (
+        generatorDefaults['unitTestRunner'] &&
+        generatorDefaults['unitTestRunner'] === 'karma'
+      ) {
+        nxJson.generators[generatorName]['unitTestRunner'] = undefined;
+      }
```

Comment:
Generator defaults can be nested inside a collection object. For example:

```json
{
  "@nrwl/angular": {
    "application": {
      ...
    },
    "library": {
      ...
    }
  }
}
```

We need to handle such configuration.